plugins/modules/site_workflow_manager.py

```diff
@@ -555,111 +554,147 @@ def get_current_site(self, site):
+                            self.log("Raw response from get_site: {}".format(response), "DEBUG")
+
+                            if isinstance(response, dict):
+                                response_data = response.get("response", [])
+                            elif isinstance(response, list):
+                                self.log("Unexpected list returned from get_site, skipping: {}".format(response), "ERROR")
+                                continue
+                            else:
+                                self.log("Unexpected response type: {}".format(type(response)), "ERROR")
+                                continue
+
                             if not response_data:
                                 self.log("No site information found for name: {0}".format(name), "WARNING")
-                                return (site_exists, current_site)
+                                continue
 
                             for site in response_data:
                                 if isinstance(site, dict):
                                     current_site = dict(site.items())
-                                    if site.get('nameHierarchy'):
-                                        current_site['parentName'] = site['nameHierarchy'].rsplit('/', 1)[0]
-                                    else:
-                                        current_site['parentName'] = None
-
+                                    current_site['parentName'] = site.get('nameHierarchy', '').rsplit('/', 1)[0] if site.get('nameHierarchy') else None
                                     all_sites_info.append(current_site)
-                                    site_exists = True
+
+                            site_exists = True
+
                         except Exception as e:
                             self.log("Error fetching site for name '{0}': {1}".format(name, str(e)))
 
-                    return (site_exists, all_sites_info)
+                    if all_sites_info:
+                        self.log("All site information collected from bulk operation: {}".format(all_sites_info), "DEBUG")
+                        return site_exists, current_site
 
-            except Exception as e:
-                self.log("Bulk operation is not available due to : {0}".format(str(e)), "WARNING")
-                name_hierarchy = self.want.get("site_name")
-                try:
-                    response = self.get_site(name_hierarchy)
-                except Exception as e:
-                    self.log("Error fetching site for {0}: {1}".format(name_hierarchy, str(e)))
-                response = response.get("response")
-                if not response:
-                    self.log("No site information found for name hierarchy: {0}".format(name_hierarchy), "WARNING")
-                    return (site_exists, current_site)
+                name_hierarchy = self.want.get("site_name_hierarchy")
+                response = self.get_site(name_hierarchy)
```

Comment:
can we rename to site_response instead of response

Reply:
Renamed